pkg/cache/partition_info_test.go

```diff
@@ -772,3 +772,46 @@ partitions:
 	assert.Equal(t, len(m), 2)
 	assert.Assert(t, reflect.DeepEqual(m["memory"], []int{1, 1, 0, 0, 0, 0, 0, 0, 1, 0}))
 }
+
+func TestCalculateAbsUsedCapacity(t *testing.T) {
```

Comment:
See above this belongs in `Resource` and we need to test all cases:
* nil, nil
* zero, nil
* set res, nil
* set res, zero
* set res, set res
* partial set res, set res
* set res, partial set

and overflow checks
using `t.log` is good during development but we have seen each log entry tagged as a failure when there is another failures in any test in the file when we run the test in a github workflow. Please remove them for now